File: plots/cm_vs_tmdr_or_tms.R
import::here(plot_roc_auc, .from='roc_auc.R')

default_colors = c('Test: TMD+TMR vs CM'='#E69F00', 'Test: TMS vs CM'='#56B4E9', 'Train: TMD+TMR vs CM'='#999999')


roc_auc_tmdr_cm = function(roc_auc, colors=default_colors, annotation=0.05, table_xmin=0.45, table_xmax=0.75, ...) {
    (
        plot_roc_auc(
            roc_auc, annotate_lines=T, table='full',
            color='set', linetype='models',
            random_color='grey50', random_linetype=3,
            table_row_colors=colors, annotation_at=annotation,
            annotation_x=annotation, nudge_x=1,
            nudge_y=-1,
            direction='y',
            ylim=c(0.35, 0.65),
            table_xmin=table_xmin, table_xmax=table_xmax, ...
        )
        + scale_color_manual(name='Set', values=colors)
        + scale_fill_manual(name='Set', values=colors)
        + scale_linetype_manual(
            name='Tested on',
            values=c('Cross-Validation'='dashed', 'Full model'='solid')
        )
        + guides(
            linetype=guide_legend(title.position='top'),
            fill=guide_legend(title.position='top'),
            color=guide_legend(title.position='top')
        )
    )
}

File: plots/cm_vs_tmdr_or_tms.py
from pandas import concat


def cm_vs_tmdr_or_tms_roc_curves(model, tms_test_set):

    tms_result = model.validate(tms_test_set)
    tms_result_sub = model.cross_validation.validate(tms_test_set)

    return concat([
        model.cross_validation_results.roc_auc.assign(group='CV train: TMD+TMR vs CM', models='Cross-Validation', set='Train: TMD+TMR vs CM'),
        model.sub_sampling_test_results.roc_auc.assign(group='CV test: TMD+TMR vs CM', models='Cross-Validation', set='Test: TMD+TMR vs CM'),
        tms_result_sub.roc_auc.assign(group='CV test: TMS vs CM', models='Cross-Validation', set='Test: TMS vs CM'),
        tms_result.roc_auc.assign(group='Full test: TMS vs CM', models='Full model', set='Test: TMS vs CM'),
        model.test_result.roc_auc.assign(group='Full test: TMD+TMR vs CM', models='Full model', set='Test: TMD+TMR vs CM')
    ])